app/util/Parser.scala

```diff
@@ -47,7 +47,21 @@ object Parser {
   def jsonToCapiData(json: Json): Either[ProductionMetricsError, CapiData] =
     json.as[CapiData].fold(processException, Right(_))
 
-  def jsonToForkData(json: Json): Either[ProductionMetricsError, ForkData] = {
-    json.as[ForkData].fold(processException, Right(_))
+  def extractMetricOpt(body: Option[String]): Either[ProductionMetricsError, MetricOpt] = body match {
```

Comment:
There are a couple of aspects of this function that feel a bit too specific given that is it now in the Parser. I would consider renaming body and making the `Invalid Json Error` message a bit more general as both feel very tied to converting the body of a request where as this function is just a parser for converting a string to a `MetricsOpt`

The same is true for `extractForkData`